hawkeye-api/src/backend.rs

```diff
@@ -0,0 +1,431 @@
+                        ),
+                        created_at: Some(
+                            labels
+                                .get("hawkeye.api.user.created_at")
+                                .unwrap_or(&default_label)
+                                .to_string(),
+                        ),
+                        updated_at: Some(
+                            labels
+                                .get("hawkeye.api.user.updated_at")
+                                .unwrap_or(&default_label)
+                                .to_string(),
+                        ),
+                        username,
+                        password: None,
+                        password_hash: Some(password),
+                        role: labels
+                            .get("hawkeye.api.user.role")
+                            .unwrap_or(&default_label)
+                            .parse()
+                            .unwrap(),
+                        owner_id: None,
+                    }
+                })
+                .collect::<Vec<User>>())
+        }
+        Err(err) => return Err(eyre!("Problem in paradise: {:?}", err)),
+    }
+}
+
+fn user_to_secret(user: &mut User) -> Secret {
+    Secret {
+        data: None,
+        immutable: Some(false),
+        metadata: ObjectMeta {
+            name: Some(templates::user_secret_name(
+                &user.id.as_ref().unwrap().to_string(),
+            )),
+            labels: Some(BTreeMap::from([
+                ("app".to_string(), "hawkeye".to_string()),
+                (
+                    "hawkeye.api.user.id".to_string(),
+                    user.id.as_ref().unwrap().to_string(),
+                ),
+                ("hawkeye.api.user.role".to_string(), user.role.to_string()),
+                (
+                    "hawkeye.api.user.created_at".to_string(),
+                    user.created_at.as_ref().unwrap().to_string(),
+                ),
+                (
+                    "hawkeye.api.user.updated_at".to_string(),
+                    user.updated_at.as_ref().unwrap().to_string(),
+                ),
+            ])),
+            ..ObjectMeta::default()
+        },
+        // This is a write-only convenience field. These get encoded to the `data` attribute.
+        string_data: Some(BTreeMap::from([
+            (
+                "hawkeye.api.user.username".to_string(),
+                user.username.to_string(),
+            ),
+            (
+                "hawkeye.api.user.password_hash".to_string(),
+                // Take password_hash if supplied (user fixture case), otherwise (api case)
+                // take/hash user.password
+                user.password_hash.as_ref().unwrap().to_string(),
+            ),
+        ])),
+        type_: Some("Opaque".to_string()),
+    }
+}
+
+/// Create a User by serializing it to a Kubernetes Secret.
+pub async fn create_user(user: &mut User, k8s_client: &kube::Client) -> Result<User, Report> {
```

Comment:
I think it would be better to accept a `User` here, and return it at the end as well:

```suggestion
pub async fn create_user(user: User, k8s_client: &kube::Client) -> Result<User, Report> {
```